Update amendments worked example (#950)
open-contracting/standard: standard/docs/en/implementation/amendments.md

* **The amendments** building block. This can contain an array of amendment explanations, and clearly identify the releases that contain before and after values. 

## Worked examples

> Each example includes sample JSON files located at the bottom of the page. Use the combobox at the end of the page to display the files for each example and step.

Reviewer: I think it would be better to thread the sample JSON files in between each step, as with the [easy releases example](https://standard.open-contracting.org/docs-easy-releases/en/implementation/easy-releases/) and use expanding/collapsing to highlight the relevant sections.

> ### Example 1: Tender

Reviewer: ```suggestion
### Example 1: Tender updates and amendments
```

This example goes through updates during the **tender** stage in a contracting process.

1. A tender is issued for a "Data merge tool" in a release with the `tag` 'tender'. 
2. Information is updated in a new release with the `tag` 'tenderUpdate' to provide an expanded description of the item. The publisher did not consider this to be a formal 'amendment' to the tender, so did not publish any amendment information.

> 3. Then there is a new update, where the value of the tender increases and the deadline for bid submissions is extended. This is considered as an 'amendment' by the publisher (depending of jurisdiction, both changes require an official approval), and so the new release has the `tag` 'tenderAmendment' and an `amendments` block under `tender`. The release reflects the updated value (USD 2000 instead of USD 1000) and the updated closing date for bid submissions (`2012-02-20` instead of `2012-02-15`).

Reviewer: depending of jurisdiction, both changes require an official approval -> depending on jurisdiction, certain changes can need to be disclosed as amendment

Reviewer: Outstanding typo:

'depending of jurisdiction' > 'depending on jurisdiction'

Note in this example that:

> * **The amendments block does not contain amended values (as in previous versions of OCDS)**. Values are updated in the same fields and using a new release. 

Reviewer: ```suggestion
* **The amendments block does not contain data on what was changed**. Changes are recorded by updating the properties of the `tender` block in a new release. 
```

> * **The publisher chooses in the tenderAmendment release to repeat a fragment of the original 'tender' release**. This is not necessary when a full archive of releases is made accessible, but can be a convenient way of publishers making structured information on the before and after values of a field more accessible to users. 

Reviewer: Add quotes to tenderAmendment

Reviewer: > but can be a convenient way of publishers making structured information on the before and after values of a field more accessible to users. 

Unless I misunderstood, a amendment release should not contain the before values?

Author: This is present in the current text. I don't think it means to say the amendment release contains the before values, but that it might be easier to users to have full blocks of data to compare. I'm not sure this is true though, so I will change the sentence.

Reviewer: Great, thanks!

> * **In the merged record**, the information in the `amendments` array can be used to lookup information in the `versionedRelease` section, to see where changes are explained by an amendment `rationale`. 

Reviewer: Please update `versioned-tender.json` to reflect the changes in `merge-tender-3.json`

> * **In the merged record**, the information in the `amendments` array can be used to lookup information in the `versionedRelease` section, to see where changes are explained by an amendment `rationale`. 

What information from the amendments array would be looked up in the versionedRelease?

Author: This was in the original example as well. `releaseID` and `amendsReleaseID` can be used to find changes in the `versionedRelease`, I'll clarify this in the text.

Reviewer: Do you think that it would make more sense to say that `releaseID` and `amendsReleaseID` can be used to find the releases with the before and after values in the `releases` array?

I am wondering which is easier from a user's perspective: checking through every field in the versioned release to find where both `releaseID` and `amendsReleaseID` appear, or comparing the two releases in the releases array.

Reviewer: Did you have a view on this?

If we don't know which is easier, we can just update the sentence to mention both, e.g.

> * **In the merged record**, the information in the `amendments` array can be used to lookup information in the `versionedRelease` section or in other releases, to see where changes are explained by an amendment `rationale`.

> ### Example 2: Contract

Reviewer: ```suggestion
### Example 2: Contract amendment
```

> This example shows an update in the value and scope of a contract.

Reviewer: ```suggestion
This example shows an update to the value and scope of a contract.
```

1. A contract notice is published for the purchase of domestic appliances. The release uses the 'contract' `tag`.

> 2. The scope of the contract is increased to include the purchase of one additional appliance. A single item is added in the `contracts/items` block and the value of the contract is increased.

Reviewer: Explain that a 'contractUpdate' release is published.

See the example release below.

```eval_rst

.. jsoninclude:: ../_static/json/amendments-contract-example.json
   :jsonpointer: /records/0/releases/1
   :expand: tag, contracts, items, amendments
   :title: ContractAmendment

```

Note that amendments can cover more than values or duration. Also, note that the publisher chose to not repeat the contract items, but add a new one with a new ID value. 

In certain scenarios there may not be a valid `amendsReleaseID` and so it can be omitted, e.g. when historical data is being published in a single release.

#### Record

An example record for the whole process is shown below. Consider downloading the <a href="../../_static/json/amendments-contract-example.json" target="_blank">record example</a> and use the [Data Review Tool](https://standard.open-contracting.org/review) to explore the changes in the contracting process.

Note that the `compiledRelease` contains all the items, included the latest one added in the contract amendment.

```eval_rst

.. jsoninclude:: ../_static/json/amendments-contract-example.json
   :jsonpointer: 
   :expand: records, releases
   :title: Record

```

### Example 3: Amendments in a Easy Releases scenario

> The Easy releases guidance (<mark>add link</mark>) explains how to build releases when there is no access to the change history for contracting processes. In a scenario when little or no change history is present, it may or may not be possible to identify and build amendments inside releases.

Reviewer: ```suggestion
The Easy releases guidance (<mark>add link</mark>) explains how to publish releases without storing or publishing a full change history. Depending on the source system, it may still be possible to publish a history of amendments when using this model.
```

> However, it is possible for a publisher to store partial history of events that may be relevant to them, like contract amendments. This example shows how amendment releases can be built when amendment history is present.

Reviewer: ```suggestion
Where the source system stores a history of contract amendments, either as separate notices or as properties of the original contract notice, contract amendments can be published as separate releases in OCDS.
```

> A publisher does not register changes in the contracting process. However, since contract amendments are a more frequent ocurrence, these are stored as new contract notices with a reference to the contract being modified. 

Reviewer: I think this is now covered by the suggested change above, but it would be good to replace it with a link to a real life example.

[CN3350299](https://www.tenders.gov.au/Cn/Show/03a3c53e-b3bd-eac1-558a-4c659e44a516) from AusTender has many amendments.

We can add sentence to make it clear that if this was published using only a single release in OCDS, then only the latest contract value and duration would be available to users, however if each amendment is published as a separate release then the full history of changes to the value and duration is available.
